docs/_docs/admin-guide/admin-installation.md

```diff
@@ -48,7 +48,39 @@ docker run -it --rm -v /var/run/docker.sock:/var/run/docker.sock
 Alternatively, you can edit the `CODENVY_HOST` value in `codenvy.env`. 
 
 ## Ports
-TODO - UPDATE PORT DESCRIPTION SECTION
+The master node is where Codenvy is installed and running. In a [scalability mode]({{base}}/docs/admin-guide/managing/index.html), you can add additional physical "workspace" nodes to increase system capacity. If you have not added any additional physical workspace nodes, then the Codenvy master node doubles as a workspace node and needs both sets of ports opened.
+
+#### Master Node: External
+|Port|Service|Notes|
+|---|---|---|
+|80 / 443|HAProxy HTTP/S|HTTP is the default. If you configure [HTTP/S]({{base}}/docs/admin-guide/configuration/index.html#https), then port 80 can be closed.
+|5000|Docker Registry|Embedded registry to save workspace snapshots. This port is not required if you configure an external registry.
+
+#### Master Node: Internal
+|Port|Service
+|---|---|
+|81|Nginx
+|2375|Swarm
+|2376|ZooKeeper
+|5432|Postgres
+|8080|Codenvy Server
+
+#### Workspace Node: External
+|Port|Service|Notes|
+|---|---|---|
+|80 / 443|HAProxy HTTP/S|HTTP is the default. If you configure [HTTP/S]({{base}}/docs/admin-guide/configuration/index.html#https), then port 80 can be closed.
+|32768-65535|Docker|Codenvy users who launch servers in a workspace have outputs mapped to this range. This range can be limited.
+
+The Docker daemon will need to be remotely accessed by Codenvy, so it has to be [setup to use a TCP socket](https://docs.docker.com/engine/reference/commandline/dockerd/#/daemon-socket-option). This port only needs to be accessible to the Codenvy master node.
+
+#### Workspace Node: Internal
+|Port|Service
+|---|---|
+|81|Nginx
+|2375|Swarm
+|2376|ZooKeeper
```

Comment:
Zookeeper, once run, will be running on a master node, and Docker daemon running on a machine node will connect to it.